CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/ViewModels/BaseLineViewModel.cs

```diff
@@ -78,88 +52,158 @@ private void fillLayout()
             ;
             for (int i = m_counterID; i < m_counterID + 3; i++)
             {
-                if (m_GridToMake.Children.Count < 5)
-                {
-                    int status = -1;
-                    buttonToMake = new Button();
-                    buttonToMake.Padding = new Thickness(20, 20, 20, 20);
-                    buttonToMake.Text = i.ToString();
-                    buttonToMake.Clicked += Button_Clicked;
-                    buttonToMake.CornerRadius = 10;
-                    buttonToMake.BackgroundColor = Color.Gold;
-                    OrderIDNotifier m = new OrderIDNotifier(i, status);
-                    m_GridToMake.Children.AddVertical(buttonToMake);
-                    order = new OrderItemView(nmList[nameIdx++], buttonToMake, m);
-
-                    m.PropertyChanged += M_PropertyChanged;
-                    m_Orders.Add(i, order);
-                }
-                else
-                {
-                    // add to waiting list 
-                }
+
+                int status = 0;
+                buttonToMake = new Button();
```

Comment:
It seems wasteful to create a new button here, only to copy it's details into an existing button in an OrderButtonModel instance. Is it not better to just be able to add this button you just made as a reference into the instance?

Reply:
maybe , but we need two way data binding in that case , keeping a ref to that button was necessary